### PR TITLE
Call contract read methods from the client

### LINKED ISSUE #2001: Call contract read methods from the client
**Dev:**
- Remove calls to API to the following resource: `/api/v2/smart-contracts/:hash/methods-read`, `/api/v2/smart-contracts/:hash/methods-read-proxy`, `/api/v2/smart-contracts/:hash/query-read-method`, `/api/v2/smart-contracts/:hash/methods-write`, `/api/v2/smart-contracts/:hash/methods-write-proxy`
- Get contract ABI from `/api/v2/smart-contracts/:hash` and display contract methods in the corresponding tabs; for proxy-contract also requests ABI if its implementation(s)
- [x] If there is no public RPC node for a blockchain, the client should make all requests via the Blockscout proxy node (need an example)

**Design:**
- [x] For methods that return a constant value show a "Retry" button when a query fails. Also, add a button to re-request the displayed value.
- [x] In the proxy tabs, we need a selector for contract implementations. This will allow users to choose methods from the desired contract.

**Testing:**
- [ ] Deploy a contract where we can test a major part of all corner cases (e.g with complex method arguments structure)

### PR DESCRIPTION
**Pull request by tom2drum**
## Description and Related Issue(s)

Resolves #2001, #1655

## Checklist for PR author
- [x] I have tested these changes locally.
- [x] I added tests to cover any new functionality, following this [guide](./CONTRIBUTING.md#writing--running-tests)
- [x] Whenever I fix a bug, I include a regression test to ensure that the bug does not reappear silently.
- [x] If I have added, changed, renamed, or removed an environment variable, I have updated the list of environment variables in the [documentation](ENVS.md) and  made the necessary changes to the validator script according to the [guide](./CONTRIBUTING.md#adding-new-env-variable)

### CODE REVIEW

File: ui/address/contract/methods/form/resultPublicClient/ItemPrimitive.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import { matchInt } from '../utils';
+import ItemLabel from './ItemLabel';
+import { printRowOffset } from './utils';
+
+function castValueToString(value: unknown): string {
+  switch (typeof value) {
+    case 'string':
+      return value === '' ? `""` : value;
+    case 'boolean':
+      return String(value);
```
Comment (isstuev) [inline ui/address/contract/methods/form/resultPublicClient/ItemPrimitive.tsx:20]: it's default case

**Review by isstuev — APPROVED** [review]
